File: app/src/main/java/pl/mareklangiewicz/myhub/di/ApplicationComponent.java
package pl.mareklangiewicz.myhub.di;

import android.app.Application;
import android.content.Context;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Component;
import pl.mareklangiewicz.myhub.MGApplication;
import pl.mareklangiewicz.myhub.ui.MyAccountFragment;
import pl.mareklangiewicz.myhub.ui.MyReposFragment;

@Singleton
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {

    @Named("Application") Context applicationContext();

    Application application();

    void inject(MGApplication a);
    void inject(MyAccountFragment f);
    void inject(MyReposFragment f);
}

File: app/src/main/java/pl/mareklangiewicz/myhub/di/ApplicationComponent.kt
package pl.mareklangiewicz.myhub.di

import android.app.Application
import android.content.Context

import javax.inject.Named
import javax.inject.Singleton

import dagger.Component
import pl.mareklangiewicz.myhub.MHApplication
import pl.mareklangiewicz.myhub.ui.MyAccountFragment
import pl.mareklangiewicz.myhub.ui.MyReposFragment

@Singleton
@Component(modules = arrayOf(ApplicationModule::class))
interface ApplicationComponent {

    @Named("Application") fun applicationContext(): Context

    fun application(): Application

    fun inject(a: MHApplication)
    fun inject(f: MyAccountFragment)
    fun inject(f: MyReposFragment)
}
